### PR TITLE
MAKE-251: add authentication to cedar application

### PR DESCRIPTION
**Pull request by julianedwards**

### CODE REVIEW

File: config.go [hunk 1]
```diff
@@ -15,21 +16,25 @@ type Configuration struct {
 	MongoDBDialTimeout time.Duration
 	UseLocalQueue      bool
 	NumWorkers         int
+	UserManager        gimlet.UserManager
```
Comment (tychoish) [inline config.go]: we can't really put this here... interfaces don't seralize well, and this config object is really about bootstrapping the thing in model.Config

File: environment.go [hunk 2]
```diff
@@ -23,6 +25,8 @@ func resetEnv() { globalEnv = &envState{name: "global", conf: &Configuration{}}
 type Environment interface {
 	Configure(*Configuration) error
 
+	AddAuth(gimlet.UserManager, time.Duration)
```
Comment (tychoish) [inline environment.go]: expire after should be configurable in the DB config struct.

Comment (tychoish) [reply environment.go]: (also in general, given a config struct or a subset of it, the production of a user manager should be lightweight and not the kind of thing that we need to cache like this.

File: operations/service.go [hunk 3]
```diff
@@ -108,7 +110,33 @@ func Service() cli.Command {
...
+					middlewareConf,
+				)
+				stream := aviation.MakeAuthenticationRequiredStreamingInterceptor(
+					cedarConf.UserManager,
+					middlewareConf,
+				)
+				rpcSrv = grpc.NewServer(
+					grpc.UnaryInterceptor(unary),
+					grpc.StreamInterceptor(stream),
+				)
```
Comment (tychoish) [inline operations/service.go]: I'd just merge all these statements and do it inline,

File: operations/service.go [hunk 4]
```diff
@@ -108,7 +110,33 @@ func Service() cli.Command {
 			//
 			// starting grpc
 			//
-			rpcSrv := grpc.NewServer()
+			middlewareConf := gimlet.UserMiddlewareConfiguration{
+				// TODO: figure out correct input for these
+				CookieName:     "cookieToken",
```
Comment (tychoish) [inline operations/service.go]: mirror what MCI uses

File: rest/data/perf.go [hunk 5]
```diff
@@ -48,8 +48,7 @@ func (dbc *DBConnector) FindPerformanceResultsByTaskId(taskId string, interval u
 			TaskID: taskId,
 			Tags:   tags,
 		},
-		// set MaxDepth to -1 to avoid recursive child search
-		MaxDepth: -1,
```
Comment (tychoish) [inline rest/data/perf.go]: doesn't this revert a thing that you changed last week?

Comment (julianedwards) [reply rest/data/perf.go]: No I forgot to change this with those changes (remember -1 now means to recurse without limit and 0 means do not recurse at all)

File: operations/service.go (hunk 3 above)
Comment (julianedwards) [reply operations/service.go]: done

**Pull request merged**